src/llmcompressor/utils/helpers.py

```diff
@@ -1132,9 +1132,39 @@ def calibration_forward_context(model: PreTrainedModel):
 
 
 @contextlib.contextmanager
-def preserve_attr(base: object, attr: str):
-    value = getattr(base, attr)
+def patch_attr(base: object, attr: str, *args, **kwargs):
```

Comment:
why use *args and **kwargs here? It seems like you can just have a `patch_value: Any = None` input rather than trying to infer based on `len(args)` or `"value" in kwargs`

Reply:
It's a little pedantic, but this allows the user to pass `None` as a valid replacement value. Otherwise, there's no way to tell the difference between a user-passed `None` and a default `None`

Reply:
That might be confusing for a user. Sounds like we want to use a sentinel value to distinguish user-passed-in None vs a default None. Have you heard of the concept? Here's a PEP that kinda explains it -- https://peps.python.org/pep-0661

Reply:
Yeah, using a sentinel or custom type definition are both good, although my personal taste is to have `patch_attr` match the same function signature as `getattr`

Reply:
Any are good

Reply:
So desired behavior is if a user explicitly passes in none, we want
```python
base[attr]=None
```
and if a user doesn't pass in anything, we want
```python
del base[attr]
```
?

Reply:
@brian-dellabetta No, the value is not deleted, only restored upon exit

Reply:
I realize that patch_attr with no argument doesn't really have a use case, so I removed that functionality for the sake of clarity and ease of use